lib/error-handler.js

```diff
@@ -62,13 +64,31 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const result = func(error, reply.request, reply)
-  if (result !== undefined) {
-    if (result !== null && typeof result.then === 'function') {
-      wrapThenable(result, reply)
-    } else {
-      reply.send(result)
+  try {
+    const result = func(error, reply.request, reply)
+    if (result !== undefined) {
+      if (result !== null && typeof result.then === 'function') {
+        wrapThenable(result, reply)
+      } else {
+        reply.send(result)
+      }
+    }
+  } catch (err) {
+    reply[kReplyIsError] = true
+    try {
+      if (err.cause) {
+        throw FST_ERR_ERROR_CAUSE_ALREADY_PRESENT()
+      }
+      err.cause = error
+    } catch (failedToAssignError) {
+      reply.log.warn({
+        err: error,
+        parentError: err,
+        reason: failedToAssignError,
+        message: 'Failed to assign child error as cause to parent error thrown from error handler'
+      })
```

Comment:
It's not clear to me why we don't:

```js
if (Object.prototype.hasOwnProperty.call(err, 'cause') === true) {
  err = Error('something or other', { cause: error })
} else {
  err.cause = error
}
```

Instead of a second `try/catch` and adding a log people are going to miss (and likely won't want, resulting in new issues being filed when they do notice it).

Reply:
@jsumners 

His claim is, that if Error Object is frozen or differently protected, you can not set the attribute.

Reply:
> That is extremely unlikely to happen, and probably a bug if it does.

The whole point of this function is to handle errors, i.e. cases where things aren't going as expected. I'd argue this is exactly the place to be robust to edge cases, especially because if we do throw an error here it crashes the entire server.

Reply:
Can you provide an example of anything generating and throwing a frozen error object?